### PR TITLE
Update EngineFinder.cmake

### PR DESCRIPTION
**Pull request by Shatur**
`EngineFinder.cmake` was changed in upstream. Current version appends a dictionaries like array values to `CMAKE_MODULE_PATH`. It results in the path like the following:

```
{
  "path" : "D:/Git/o3de"
}/cmake
```

The PR fixes the issue. `EngineFinder.cmake` was taken from the upstream.

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]

File: EngineFinder.cmake [hunk 1]
```diff
@@ -1,12 +1,9 @@
-# {BEGIN_LICENSE}
```
Comment (Shatur) [inline EngineFinder.cmake:1]: This line appears when I creating a new project. Is this correct?

Comment (amznestebanpapp) [reply EngineFinder.cmake:1]: looks like is part of the template, however, I dont see a need to have that, if you create an issue in o3de I can remove it from the template

**Review by amznestebanpapp — APPROVED** [review]

**Comment by lumberyard-employee-dm** [comment]
@Shatur your commit has failed the digital certificate of origin check.
In order to submit to the O3DE repo you are going to need to amend the commit to signoff on it using the `--signoff` to git commit
https://github.com/o3de/o3de-multiplayersample/pull/42/checks?check_run_id=3175711687

**Comment by Shatur** [comment]
@lumberyard-employee-dm, thanks, done!

**Pull request merged**